tests/models/test_taskinstance.py

```diff
@@ -5290,3 +5290,70 @@ def test_swallow_mini_scheduler_exceptions(_schedule_downstream_mock, create_tas
     ti.schedule_downstream_tasks()
     assert "Error scheduling downstream tasks." in caplog.text
     assert "To be swallowed" in caplog.text
+
+
+def test_ti_selector_condition():
+    from airflow.decorators import dag, task_group
+    from airflow.models.taskinstance import TaskInstance
+    from airflow.operators.empty import EmptyOperator
+    from airflow.utils.session import create_session
+    from airflow.utils.state import State
+    from airflow.utils.timezone import datetime
+    from airflow.utils.types import DagRunType
+    from tests.test_utils.db import clear_db_runs
```

Comment:
Why do we need to add these imports here?

Reply:
I used `dag_maker` as you suggested

Reply:
Any other changes I need to make? 
Hopefully the logic itself is correct now?